pkg/Cpanel/Security/Advisor/Assessors/Trojans.pm

```diff
@@ -73,7 +92,521 @@ sub _check_for_libkeyutils {
+                'Check the following for more information "[output,url,_1,We Live Security More Information,_2,_3]"',
+                'http://www.welivesecurity.com/wp-content/uploads/2014/03/operation_windigo.pdf',
+                'target',
+                '_blank'
+            ],
+        );
+    }
+}
+
+sub build_libkeyutils_file_list {
+    my ($self) = @_;
+    my @dirs = qw( /lib /lib64 );
+    my @libkeyutils_files;
+
+    for my $dir (@dirs) {
+        next if !-e $dir;
+        opendir( my $dir_fh, $dir );
+        while ( my $file = readdir($dir_fh) ) {
+#            if ( $file =~ /^libkeyutils\.so\.(?:[\.\d]+)?$/ ) {
+            if ( $file =~ /^libkeyutils\.so(?:\.[\.\d]+)?$/ ) {
+                if ( $dir eq '/lib' ) {
+                    push @libkeyutils_files, "/lib/$file";
+                }
+                elsif ( $dir eq '/lib64' ) {
+                    push @libkeyutils_files, "/lib64/$file";
+                }
+            }
+        }
+        closedir $dir_fh;
+    }
+
+    return \@libkeyutils_files;
+}
+
+sub _check_sha1_sigs_libkeyutils {
+    my ($self) = @_;
+    return if !$LIBKEYUTILS_FILES_REF;
+
+    my $trojaned_lib;
+
+    # p67 http://www.welivesecurity.com/wp-content/uploads/2014/03/operation_windigo.pdf
+    my @checksums = qw(
+      09c8af3be4327c83d4a7124a678bbc81e12a1de4
+      1a9aff1c382a3b139b33eeccae954c2d65b64b90
+      267d010201c9ff53f8dc3fb0a48145dc49f9de1e
+      2e571993e30742ee04500fbe4a40ee1b14fa64d7
+      2fc132440bafdbc72f4d4e8dcb2563cc0a6e096b
+      39ec9e03edb25f1c316822605fe4df7a7b1ad94a
+      3c5ec2ab2c34ab57cba69bb2dee70c980f26b1bf
+      471ee431030332dd636b8af24a428556ee72df37
+      58f185c3fe9ce0fb7cac9e433fb881effad31421
+      5d3ec6c11c6b5e241df1cc19aa16d50652d6fac0
+      74aa801c89d07fa5a9692f8b41cb8dd07e77e407
+      7adb38bf14e6bf0d5b24fa3f3c9abed78c061ad1
+      899b860ef9d23095edb6b941866ea841d64d1b26
+      8daad0a043237c5e3c760133754528b97efad459
+      8f75993437c7983ac35759fe9c5245295d411d35
+      9bb6a2157c6a3df16c8d2ad107f957153cba4236
+      9e2af0910676ec2d92a1cad1ab89029bc036f599
+      a7b8d06e2c0124e6a0f9021c911b36166a8b62c5
+      adfcd3e591330b8d84ab2ab1f7814d36e7b7e89f
+      b8508fc2090ddee19a19659ea794f60f0c2c23ff
+      bbce62fb1fc8bbed9b40cfb998822c266b95d148
+      bf1466936e3bd882b47210c12bf06cb63f7624c0
+      d552cbadee27423772a37c59cb830703b757f35e
+      e14da493d70ea4dd43e772117a61f9dbcff2c41c
+      e2a204636bda486c43d7929880eba6cb8e9de068
+      f1ada064941f77929c49c8d773cbad9c15eba322
+    );
+
+    for my $lib (@$LIBKEYUTILS_FILES_REF) {
+        next unless my $checksum = _check_sha1sum($lib, @checksums);
+        $trojaned_lib = "$lib\n\tSHA-1 checksum: $checksum";
+        last;
+    }
+
+    if ($trojaned_lib) {
+        $self->add_bad_advice(
+            'text'       => [ "Libkey rootkit check: suspicious checksums/hashes [_1] were found that could indicate the existence of the Ebury rootkit.", $trojaned_lib ],
```

Comment:
The output of this (and several other) advice may not look good when rendered.  Consider the output if 2 files are found to have the issue:

```
Libkey rootkit check: suspicious checksums/hashes /lib64/libkeyutils.so.6
    SHA-1 checksum: f1ada064941f77929c49c8d773cbad9c15eba322/lib64/libkeyutils.so.6
    SHA-1 checksum: e2a204636bda486c43d7929880eba6cb8e9de068were found that could indicate the existence of the Ebury rootkit.
```

In general, we add the list of items after the statement, as that avoids broken lines and makes listing more natural.